### DISCUSSION TITLE
Using Rosetta2 with Fedora; "update-binfmts" is not available

### DISCUSSION THREAD

**Discussion by alshdavid** [utmapp/UTM#4601] [UTM Help (macOS)]
When following the [instructions][1] provided by Apple and UTM, we need to use `/usr/sbin/update-binfmts` to (I assume) register Apple's Rosetta2 binary to handle the translation of x86 binaries when they are executed.

When trying to run the command to register `rosetta` as the handler, Fedora tells me that `update-binfmts` is not found. 

```bash
$ sudo /usr/sbin/update-binfmts --install rosetta /media/rosetta/rosetta \
     --magic "\x7fELF\x02\x01\x01\x00\x00\x00\x00\x00\x00\x00\x00\x00\x02\x00\x3e\x00" \
     --mask "\xff\xff\xff\xff\xff\xfe\xfe\x00\xff\xff\xff\xff\xff\xff\xff\xff\xfe\xff\xff\xff" \
     --credentials yes --preserve no --fix-binary yes

$ sudo: /usr/sbin/update-binfmts: command not found
```

I am not sure what `update-binfmts` does, but I guess this is Debian specific? 
What would the equivalent command be in Fedora to setup Rosetta?

  [1]: https://docs.getutm.app/advanced/rosetta/

**Comment by osy** [comment]
> You will need the update-binfmts command installed which is usually part of the binfmt-support package.

Not sure what it’s called in fedora but you can probably find it. Note that fedora doesn’t have arm64->x86_64 multiarch support so you’ll have to do a lot more work to get shared libraries working.

**Reply by alshdavid** [reply]
I see, would I be better off with Debian or Ubuntu? I will be installing the server variant as I don't need a GUI

**Reply by osy** [reply]
Debian as that’s why I wrote the guide for debian

**Comment by timnoack** [comment]
You can directly register a handler to the binfmt_misc kernel module in any linux distro by running as root:
```echo ":name:type:offset:magic:mask:interpreter:flags" > /proc/sys/fs/binfmt_misc/register```
For rosetta at `/media/rosetta/rosetta` this would be 
```echo ":rosetta:M::\x7fELF\x02\x01\x01\x00\x00\x00\x00\x00\x00\x00\x00\x00\x02\x00\x3e\x00:\xff\xff\xff\xff\xff\xfe\xfe\x00\xff\xff\xff\xff\xff\xff\xff\xff\xfe\xff\xff\xff:/media/rosetta/rosetta:CF" > /proc/sys/fs/binfmt_misc/register```

This needs to be run as root. You also need to run this again after each startup! Deregistering a handler is as easy as running:
```echo -1 >  /proc/sys/fs/binfmt_misc/rosetta```
